src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+
+.. sourcecode:: bash
+
+   Error:(14, 9) java: cannot find symbol
+   symbol:   variable celsius
+   location: class TempConverter
+
+   Error:(15, 64) java: cannot find symbol
+   symbol:   variable celsius
+   location: class TempConverter
+
+These two *compiler errors* occur before the program runs. The values in the
+parentheses ``()`` give the line number and text column where the error was
+found. In the first description (line 14, column 9), the ``celsius`` variable
+before the ``=`` is flagged. When this type of error happens, it usually means
+that the variable was not declared before we tried to initialize it with a
+value.
+
+The second error message (line 15, column 64) occurs because we use
+``celsius`` before it has been assigned a value.
+
+.. admonition:: Note
+
+   When using an IDE such as IntelliJ, your work is typically checked by the
+   IDEs built-in compiler as you write your code. Errors are often visually
+   indicated by the IDE as you type. This avoids having to explicitly
+   compile your code before finding errors. Nice, huh?
+
+   .. figure:: figures/IDE-flagged-errors.png
+      :alt: The ``celsius`` variables are flagged.
+
+      The red coloring of the ``celsius`` variables indicate errors.
+
+The general rule in Java is that you must decide what kind of object your
+variable is going to reference and then declare that variable before you use
+it. There is much more to say about the static typing of Java, but for now this
+is enough.
+
+.. admonition:: Note
+
+   As in other languages, Java allows you to declare and initialize your
+   variables in the same line:
+
+   .. sourcecode:: java
+
+      double celsius = (fahrenheit - 32) * 5/9;
+
+Add Comments to Your Code
+--------------------------
+
+As programs get bigger and more complicated, they get more difficult to read.
+Good programmers try to make their code understandable to others, but it is
+still tricky to look at a large program and figure out what it is doing and
+why.
+
+Also, there are times when programmers need to isolate or ignore certain
+portions of their code as they are testing it. In the "Try It" box above, you
+were instructed to *remove* a line of code in order to create compiler errors.
+However, programmers are usually reluctant to delete lines that they might need
+to bring back.
+
+.. index:: ! comments
+
+Best practice encourages us to add **comments** to our programs. These are
+notes that clearly explain what the code is doing.
+
+A comment is text within a program intended only for a human reader—--it is
+completely ignored by the compiler. In Java, the ``//`` token indicates the
+start of a comment, and the rest of the line gets ignored. For comments that
+stretch over multiple lines, the text falls between the symbols ``/*   */``.
+
+Comments can also be used to temporarily skip a portion of the code when a
+program runs. Instead of removing line 6 in ``TempConverter``, we could
```

Comment:
does line 6 refer to the celsius initialization? its kind of tough to remember that while reading the page since the code block isn't in view.

Reply:
Good call.  Fixed.